smart_card_connector_app/src/chrome-api-jstocxxtest.js

```diff
@@ -846,14 +846,63 @@ goog.exportSymbol('testChromeApiProviderToCpp', {
       expectReportPlainResult(/*requestId=*/ 111, 'SUCCESS');
 
       // TAG_IFD_DEVICE_REMOVED is used with a value of 0 here. This is a no-op.
-      // Nothing will happen to the reader. 
+      // Nothing will happen to the reader.
       // The goal is just to test `SetAttrib` returning the `SUCCESS` value.
       await mockChromeApi
           .dispatchEvent(
               'onSetAttribRequested', /*requestId=*/ 111, readerHandle,
               TAG_IFD_DEVICE_REMOVED, /*data=*/ new Uint8Array(['0x0']))
           .$waitAndVerify();
     },
-  }
+  },
+
+  // Test Transmit success.
+  'testTransmit_success': async function() {
+    // Set up a reader and a card with a PIV profile.
+    await launchPcscServer(/*initialDevices=*/[{
+      'id': 123,
+      'type': SimulationConstants.GEMALTO_DEVICE_TYPE,
+      'cardType': SimulationConstants.COSMO_CARD_TYPE,
+      'cardProfile': SimulationConstants.CHARISMATHICS_PIV_TYPE
+    }]);
+
+    const sCardContext = await establishContext(/*requestId=*/ 66);
+    readerHandle = await connect(
+        /*requestId=*/ 67, sCardContext,
+        SimulationConstants.GEMALTO_PC_TWIN_READER_PCSC_NAME0,
+        chrome.smartCardProviderPrivate.ShareMode.SHARED,
+        /*preferredProtocols=*/ {'t1': true},
+        /*expectedProtocol=*/ chrome.smartCardProviderPrivate.Protocol.T1);
+
+    // Reply should contain the application identifier, followed by the status
+    // bytes that denote success: SW1=0x90, SW2=0x00. This corresponds to the
+    // fake reply built by `testing_smart_card_simulation.cc`.
+    expectReportDataResult(
+        /*requestId=*/ 111,
+        new Uint8Array([
+          0x61, 0x5C, 0x4F, 0x0B, 0xA0, 0x00, 0x00, 0x03, 0x08, 0x00, 0x00,
```

Comment:
Let's create some constant for this. E.g., for the "application identifier", which would be everything except the last bytes "0x90, 0x00".

Reply:
done